src/cpp/shared/scripting/callback.h

```diff
@@ -17,6 +19,12 @@ void ironfist_lua_push(void *arg);
 void ironfist_lua_push(std::string arg);
 void ironfist_lua_pushmulti();
 
+template <typename Res>
+nonstd::optional<Res> GetLuaResult(lua_State *L, int arg);
+
+template <>
+nonstd::optional<bool> GetLuaResult(lua_State *L, int arg);
```

Comment:
The definition of the bool version of this function should go here too, in the .h file.